### PR TITLE
Add a StyledTreeEditor for Qt

### PR DESCRIPTION
**Pull request by jwiggins**
This overrides some of the machinery in `TreeEditor` for Qt so that users can control the rendering of items with a `QStyledItemDelegate`.

### CODE REVIEW

**Comment by corranwebster** [comment]
Is it possible to just fold these extra capabilities into the Qt TreeEditor?  I'd be fine with having a trait on the `TreeEditor` factory which holds a renderer that is only used by Qt and ignored by Wx.

File: traitsui/qt4/extra/styled_tree_editor.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+
+class StyledTreeNode(TreeNode):
+    """ A TreeNode to use with StyledTreeEditor
+    """
+    #: Set to True to use ItemDelegate for drawing
+    delegates_style = Bool(False)
+
+
+if __name__ == '__main__':
```
Comment (corranwebster) [inline traitsui/qt4/extra/styled_tree_editor.py]: Could you put this into the `examples` directories (not sure where is best, since it is Qt-specific.

**Comment by jwiggins** [comment]
I felt like it was too invasive to put this in the normal Qt TreeEditor.

File: traitsui/qt4/extra/styled_tree_editor.py (hunk 1 above)
Comment (jwiggins) [reply traitsui/qt4/extra/styled_tree_editor.py]: That shouldn't be a problem

**Comment by jwiggins** [comment]
CI is clean. (thanks @rahulporuri!)

**Comment by corranwebster** [comment]
I'm working on a refactor of this to integrate it into the main editor and generally make it more flexible (eg. be able to override renderers for additional columns if there are any).

**Comment by jwiggins** [comment]
Sounds good

**Pull request merged**